### ISSUE TITLE
Matching to Different Set of Controls

### ISSUE THREAD

**Issue by GordianArnav** [snap-stanford/GEARS#41]
Is there a way to match 'A+ctrl1' to 'ctrl1' cells and 'A+ctrl2'to 'ctrl2' in the same adata object?

**Comment by yhr91** [comment]
This is possible but would require some changes to the dataloader and the loss function

Specifically, in this section, for each sample perturbed cell the dataloader will randomly pick a control cell as the basal state `X`which is appended to the vector of basal states `Xs`

https://github.com/snap-stanford/GEARS/blob/df09d7ae34e90f5ef25afa389daf7c5c589e710d/gears/pertdata.py#L578-L584

You would have to update this code to choose from different control populations based on the perturbed cell. 

Additionally at the point of computing the loss across a batch:

https://github.com/snap-stanford/GEARS/blob/df09d7ae34e90f5ef25afa389daf7c5c589e710d/gears/utils.py#L418

it would need to be averaged separately for separate control populations.

It sounds straightforward but the actual implementation may end up requiring additional changes. We've been experimenting with something similar internally. Let me know if you have any other questions.
